data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexTemplateAPIWrapperFactory.java

```diff
@@ -0,0 +1,17 @@
+package org.opensearch.dataprepper.plugins.sink.opensearch.index;
+
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.dataprepper.plugins.sink.opensearch.DistributionVersion;
+
+public class IndexTemplateAPIWrapperFactory {
+    public static IndexTemplateAPIWrapper getWrapper(final IndexConfiguration indexConfiguration,
+                                                     final OpenSearchClient openSearchClient) {
+        if (DistributionVersion.ES6.equals(indexConfiguration.getDistributionVersion())) {
```

Comment:
Can we add some logic into the sink configuration code to ensure that when distribution version is ES6, we only allow the `v1` type?

Reply:
Great catch!